collect_app/src/test/java/org/odk/collect/android/permissions/PermissionsProviderTest.java

```diff
@@ -0,0 +1,141 @@
+package org.odk.collect.android.permissions;
+
+import android.Manifest;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.odk.collect.android.storage.StorageStateProvider;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PermissionsProviderTest {
+    private PermissionsChecker permissionsChecker;
+    private StorageStateProvider storageStateProvider;
+    private PermissionsProvider permissionsProvider;
+
+    @Before
+    public void setup() {
+        permissionsChecker = mock(PermissionsChecker.class);
+        storageStateProvider = mock(StorageStateProvider.class);
+    }
+
+    @Test
+    public void whenStoragePermissionsGrantedAndScopedStorageNotUsed_shouldAreStoragePermissionsGrantedReturnTrue() {
+        when(permissionsChecker.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE)).thenReturn(true);
+        when(storageStateProvider.isScopedStorageUsed()).thenReturn(false);
+        permissionsProvider = new PermissionsProvider(permissionsChecker, storageStateProvider);
+
+        assertThat(permissionsProvider.areStoragePermissionsGranted(), is(true));
+    }
+
+    @Test
+    public void whenStoragePermissionsGrantedAndScopedStorageUsed_shouldAreStoragePermissionsGrantedReturnTrue() {
+        when(permissionsChecker.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE)).thenReturn(true);
+        when(storageStateProvider.isScopedStorageUsed()).thenReturn(true);
+        permissionsProvider = new PermissionsProvider(permissionsChecker, storageStateProvider);
+
+        assertThat(permissionsProvider.areStoragePermissionsGranted(), is(true));
+    }
+
+    @Test
+    public void whenStoragePermissionsNotGrantedAndScopedStorageNotUsed_shouldAreStoragePermissionsGrantedReturnFalse() {
+        when(permissionsChecker.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE)).thenReturn(false);
+        when(storageStateProvider.isScopedStorageUsed()).thenReturn(false);
+        permissionsProvider = new PermissionsProvider(permissionsChecker, storageStateProvider);
+
+        assertThat(permissionsProvider.areStoragePermissionsGranted(), is(false));
+    }
+
+    @Test
+    public void whenStoragePermissionsNotGrantedAndScopedStorageUsed_shouldAreStoragePermissionsGrantedReturnTrue() {
+        when(permissionsChecker.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE)).thenReturn(false);
+        when(storageStateProvider.isScopedStorageUsed()).thenReturn(false);
```

Comment:
This should be `true` so that the last assertion is `true`!

Reply:
Ahh good catch, thanks!